src/Symfony/Component/Translation/Dumper/XliffFileDumper.php

```diff
@@ -176,7 +176,7 @@ private function dumpXliff2(string $defaultLocale, MessageCatalogue $messages, ?
             $metadata = $messages->getMetadata($source, $domain);
 
             // Add notes section
-            if ($this->hasMetadataArrayInfo('notes', $metadata)) {
+            if ($this->hasMetadataArrayInfo('notes', $metadata) && [] !== $metadata['notes']) {
```

Comment:
we don't do empty-value comparisons:
```suggestion
            if ($this->hasMetadataArrayInfo('notes', $metadata) && $metadata['notes']) {
```